File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.py
def solution(s):
    answer = ''
    idx = 0
    
    for ch in s:
        # 공백 처리
        if ch == ' ': 
            idx = 0
            answer += ' '
        else:
            if idx % 2 == 0:
                answer += ch.upper()
            else:
                answer += ch.lower()
            idx += 1
    
    return answer

File: 프로그래머스/1/Java/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        int wordIndex = 0;

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);

            if (c == ' ') {
                answer += " ";
                wordIndex = 0;
            } else {
                if (wordIndex % 2 == 0) {
                    answer += Character.toUpperCase(c);
                } else {
                    answer += Character.toLowerCase(c);
                }

                wordIndex++;
            }
        }

        return answer;
    }
}